### ISSUE TITLE
Keep or clean up intermediate HTML files

### ISSUE THREAD

**Issue by nucleic-acid** [nucleic-acid/namedropR#52]
Just want to document this thought while it is in my head (we can spawn a new issue if desired). Is there a reason why the HTML snippets are *"permanently"* stored in `visual_citations/`, by default, rather than in a temporary directory that gets cleaned up when the R session is killed?

If my understanding is correct, the file(s) contained in `visual_citations/` are no longer necessary once `htmltools::includeHTML()` has been called to inject the HTML into the main file.

I understand storing the QR codes locally when `include_qr` is `link` or `link_svg`. But it seems to me that `visual_citations/` can live in a temporary directory instead of saving intermediate files into the users local directory. Thoughts?

_Originally posted by @mattwarkentin in https://github.com/nucleic-acid/namedropR/issues/51#issuecomment-1152763450_

**Comment by nucleic-acid** [comment]
This has grown due to historic reasons: initially I thought users would include the HTML renderings via `<iframe>`.

Later I kept it due to the following reasons, but all is open for discussion:

- **Speed:** when building e.g. presentations with xaringan or revealJS you might include several citations. If they are rendered every time you re-render the slides this might prolong the overall building process of the slides significantly. If you know the sources you cite before starting with the slides, you can bulk-render them once in the beginning (e.g. in a separate script) and just link to them in the slides code.
- **Websites:** users might include the HTML in websites an not use `{htmltools}` for that. They might need to keep the HTML files.

Apart from that: As you can choose the output folder, you can achieve "cleanup" already via passing `tempdir()` as output path:

```
# set individual output folder
drop_name(bib, output_dir=tempdir())

# in combination with htmltools:
htmltools::includeHTML(
  drop_name(
    bib_path,
    output_dir = tempdir(),
    cite_key = "SomeAuthor2010",
    export_as = "html",
    use_xaringan = TRUE
  )
)
```

That way `includeHTML()` can include them without leading traces in your project directory.
Would that solve the issue for you?

**Comment by mattwarkentin** [comment]
Thanks for the detailed explanation, @nucleic-acid. I think `tempdir()` works for my purposes.
